conan/tools/cmake/cmakedeps/templates/__init__.py

```diff
@@ -74,29 +75,34 @@ def arch(self):
     def config_suffix(self):
         return "_{}".format(self.configuration.upper()) if self.configuration else ""
 
-    def get_target_namespace(self):
-        return get_target_namespace(self.conanfile)
-
     def get_file_name(self):
-        return get_file_name(self.conanfile)
-
-
-def get_target_namespace(req):
-    ret = req.new_cpp_info.get_property("cmake_target_name", "CMakeDeps")
-    if not ret:
-        ret = req.cpp_info.get_name("cmake_find_package_multi", default_name=False)
-    return ret or req.ref.name
-
-
-def get_component_alias(req, comp_name):
-    if comp_name not in req.new_cpp_info.components:
-        # foo::foo might be referencing the root cppinfo
-        if req.ref.name == comp_name:
-            return get_target_namespace(req)
-        raise ConanException("Component '{name}::{cname}' not found in '{name}' "
-                             "package requirement".format(name=req.ref.name, cname=comp_name))
-    ret = req.new_cpp_info.components[comp_name].get_property("cmake_target_name", "CMakeDeps")
-    if not ret:
-        ret = req.cpp_info.components[comp_name].get_name("cmake_find_package_multi",
-                                                          default_name=False)
-    return ret or comp_name
+        return get_file_name(self.conanfile, find_module_mode=self.find_module_mode)
+
+    def get_target_namespace(self, req):
+        if self.find_module_mode:
+            ret = req.new_cpp_info.get_property("cmake_module_target_name", "CMakeDeps")
+            if ret:
+                return ret
+
+        ret = req.new_cpp_info.get_property("cmake_target_name", "CMakeDeps")
```

Comment:
In the language I am familiar with, target names are not the same as the namespace. This is a little confusing for me

Reply:
You also have the possibility to declare components:
https://docs.conan.io/en/latest/reference/conanfile/tools/cmake/cmakedeps.html#properties

Reply:
Thanks for the review by the way! :)

Reply:
This is followed up by this issue https://github.com/conan-io/conan/issues/9511, right?

Reply:
Yes, exactly